infra/cifuzz/cifuzz.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module used by CI tools in order to interact with fuzzers.
+This module helps CI tools do the following:
+  1. Build fuzzers.
+  2. Run fuzzers.
+Eventually it will be used to help CI tools determine which fuzzers to run.
+"""
+
+import logging
+import os
+import shutil
+import sys
+
+import fuzz_target
+
+# pylint: disable=wrong-import-position
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import build_specified_commit
+import helper
+import repo_manager
+import utils
+
+
+def build_fuzzers(project_name, project_repo_name, commit_sha, git_workspace,
+                  out_dir):
+  """Builds all of the fuzzers for a specific OSS-Fuzz project.
+
+  Args:
+    project_name: The name of the OSS-Fuzz project being built.
+    project_repo_name: The name of the projects repo.
+    commit_sha: The commit SHA to be checked out and fuzzed.
+    git_workspace: The location in the shared volume to store git repos.
+    out_dir: The location in the shared volume to store output artifacts.
+
+  Returns:
+    True if build succeeded or False on failure.
+  """
+  if not os.path.exists(git_workspace):
+    logging.error('Invalid git workspace: %s.', format(git_workspace))
+    return False
+  if not os.path.exists(out_dir):
+    logging.error('Invalid out directory %s.', format(out_dir))
+    return False
+
+  inferred_url, oss_fuzz_repo_path = build_specified_commit.detect_main_repo(
+      project_name, repo_name=project_repo_name)
+  if not inferred_url or not oss_fuzz_repo_path:
+    logging.error('Could not detect repo from project %s.', project_name)
+    return False
+  src = os.path.dirname(oss_fuzz_repo_path)
+  oss_fuzz_repo_name = os.path.basename(oss_fuzz_repo_path)
+
+  # Checkout projects repo in the shared volume.
+  build_repo_manager = repo_manager.RepoManager(inferred_url,
+                                                git_workspace,
+                                                repo_name=oss_fuzz_repo_name)
+  try:
+    build_repo_manager.checkout_commit(commit_sha)
+  except repo_manager.RepoManagerError:
+    logging.error('Specified commit does not exist.')
+    # NOTE: Remove return statement for testing.
```

Comment:
is this a leftover? or do you mean the `return` should not be here when you test this?

Reply:
I mean that return should be commented out when testing the action.

Reply:
I don't want to block you at this point, but let's add at least a `TODO(Leo-Neat): ...` to figure out a solution that would not require any manual patching of the code. And somewhere in your tracking notes you can have an action item to "address all TODOs" in the project